x/stake/types.go

```diff
@@ -93,6 +96,28 @@ type Candidate struct {
 // Candidates - list of Candidates
 type Candidates []Candidate
 
+// Print out a candidate wit Bech32Cosmos encodings of Addresses and PubKeys
+func (c Candidate) Print() {
+
+	bech32addr, err := bech32cosmos.ConvertAndEncode("cosmosaccaddr", c.Address.Bytes())
+	if err != nil {
+		fmt.Printf("Address Bech32Cosmos Error: %s\ns", err)
+	}
+	bech32pk, err := bech32cosmos.ConvertAndEncode("cosmosvalpubkey", c.PubKey.Bytes())
+	if err != nil {
+		fmt.Printf("Validator PubKey Bech32Cosmos Error: %s\n", err)
+	}
+	fmt.Println("Candidate")
+	fmt.Printf("Status %v \n", c.Status)
+	fmt.Printf("Owner Address %s \n", bech32addr)
+	fmt.Printf("Validator PubKey %s \n ", bech32pk)
+	fmt.Printf("Assets %s \n", c.Assets)
+	fmt.Printf("Liabilities %s \n", c.Liabilities)
+	fmt.Printf("Description %s \n", c.Description)
+	fmt.Printf("ValidatorBondHeight %d", c.ValidatorBondHeight)
+	fmt.Printf("ValidatorBondCounter %d", c.ValidatorBondCounter)
```

Comment:
The default should be machine readable aka json. We can do something fun though and use the `--output=text` flag if we want non-json text (such as used in the `gaiacli keys`) but this should def not be the default. 

to do this I'd suggest that that `candidate.Print()` takes in a boolean to distinguish human vs. machine readable. This way only the query function deals with viper

Reply:
So this test 
https://github.com/cosmos/cosmos-sdk/blob/master/cmd/gaia/cli_test/cli_test.go#L38-L39

takes machine readable output and passes in via human readable interface points,

I could bechify the data in the test suite itself. How do you think we should handle it?

Reply:
what specifically do you mean by "human readable interface points" ? like the CLI flags?

I think we should build in `cech32` address types in the bottom layer of the sdk and have the hrp be excluded from the actual key bytes when saving in the store but probably included everywhere else (the cesh32Address could would have one function `(c Cesh32Address) RawAddress() []byte` or something to that effect which could be used. 

If we do this - the `cesh32` addresses will show up everywhere within the machine parsable output so everything will be non-ambiguous 

Does this kind of answer your question?

Reply:
the human readable part is like "cosmosaccaddr", the case insensitive hrp is part of the check sum so it bind the context of the address together for the data.

Someone could not just change the hrp and use the same address in a different context.

Reply:
Based on previous discussion with @jaekwon he wanted `cesh32` only at the human interface layers and not this deeply embedded but I am not sure.

Reply:
@zmanian @jaekwon alright - I'm okay with this, it just means that for all JSON machine output we will not receive the cesh32 addresses. We need a more consolidated strategy for CLI output (aka everything should probably use the `--output` flag)